lib/robots/sdr_repo/preservation_ingest/base.rb

```diff
@@ -35,7 +35,11 @@ def workflow_service
         private
 
         def moab_object
-          @moab_object ||= Stanford::StorageServices.find_storage_object(druid, true)
+          @moab_object = Stanford::StorageServices.search_storage_objects(druid, true).filter! { |moab| moab.storage_root == @primary_moab_location }
```

Comment:
i did some testing on stage tuesday and yesterday, and had to adjust this a bit to get it to work, but have a (draft) PR to this one with the changes that seemed to work based on my manual testing (using the integration test suite):  https://github.com/sul-dlss/preservation_robots/pull/250/files